test/testfixture/testfixture.go

```diff
@@ -0,0 +1,193 @@
+package testfixture
+
+import (
+	"context"
+	"testing"
+
+	"github.com/fabric8-services/fabric8-tenant/tenant"
+	"github.com/fabric8-services/fabric8-wit/resource"
+	"github.com/jinzhu/gorm"
+	errs "github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+)
+
+// A TestFixture object is the result of a call to
+//  NewFixture()
+// or
+//  NewFixtureIsolated()
+//
+// Don't create one on your own!
+type TestFixture struct {
+	info             map[kind]*createInfo
+	db               *gorm.DB
+	isolatedCreation bool
+	ctx              context.Context
+	checkFuncs       []func() error
+
+	Tenants    []*tenant.Tenant    // Tenants (if any) that were created for this test fixture.
+	Namespaces []*tenant.Namespace // Namespaces (if any) that were created for this test fixture.
+}
+
+// NewFixture will create a test fixture by executing the recipies from the
+// given recipe functions. If recipeFuncs is empty, nothing will happen.
+//
+// For example
+//     NewFixture(db, Comments(100))
+// will create a work item (and everything required in order to create it) and
+// author 100 comments for it. They will all be created by the same user if you
+// don't tell the system to do it differently. For example, to create 100
+// comments from 100 different users we can do the following:
+//      NewFixture(db, Identities(100), Comments(100, func(fxt *TestFixture, idx int) error{
```

Comment:
This section needs an update as well because the recipe functions don't exist in this repo.